lib/hypertrack.dart

```diff
@@ -53,49 +74,88 @@ class HyperTrack {
   /// In case of frequent changes, it is possible, that intermediate states
   /// could be lost, so if more real-time responsiveness is required it is
   /// recommended to use [addGeotag] for passing that data.
-  void setDeviceMetadata(Map<String, Object> data) =>
-      _pluginInterface.setDeviceMetadata(data);
+  Future<void> setMetadata(JSONObject data) {
+    return invokeSdkVoidMethod(
+        SdkMethod.setMetadata, serializeMetadata(data));
+  }
 
-  /// Enable debugging.
-  void enableDebugLogging() async {
-    const MethodChannel('sdk.hypertrack.com/handle')
-        .invokeMethod('enableDebugLogging');
+  Future<Result<Location, HyperTrackError>> addGeotag(JSONObject data) {
+    return invokeSdkMethod(SdkMethod.addGeotag, serializeGeotag(data))
+        .then((value) {
+      return deserializeLocationResult(value);
+    });
+  }
+
+  Future<Result<Location, HyperTrackError>> get location async {
+    return invokeSdkMethod(SdkMethod.getLocation).then((value) {
+      return deserializeLocationResult(value);
+    });
   }
 
   /// Triggers tracking start.
   ///
   /// This isn't always result in SDK tracking, as missing permissions or disabled
-  /// geolocation sensors could lead to a tracking outage. Use [onTrackingStateChanged]
+  /// geolocation sensors could lead to a tracking outage. Use [onTrackingChanged]
   /// stream to get the actual state details.
-  start() async {
-    await _pluginInterface.startTracking();
-  }
+  void startTracking() => invokeSdkVoidMethod(SdkMethod.startTracking);
 
   /// Stops tracking.
-  stop() async {
-    await _pluginInterface.stopTracking();
-  }
+  void stopTracking() => invokeSdkVoidMethod(SdkMethod.stopTracking);
 
   /// Allows you to use location mocking software (e.g. for development).
   ///
   /// Mock locations are ignored by HyperTrack SDK by default.
-  allowMockLocations([bool allow = false]) =>
-      _pluginInterface.allowMockLocations(allow);
+  void allowMockLocations() {
```

Comment:
Shouldn't this accept a boolean?

Reply:
Same situation as with enableDebugLogging